src/main/java/TTT/databaseUtils/CustomUserDAO.java

```diff
@@ -16,22 +16,36 @@ public class CustomUserDAO {
     private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
     private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
 
-    public boolean saveUser(CustomUser customUser){
+    public boolean saveUser(CustomUser customUser) {
+
+        if (findCustomUser(customUser.getEmail()) != null) {
+            return false;
+        }
+        if (customUser.getPassword().length() < 7) {
```

Comment:
In the controller you are checking for length of 8, here 7

Reply:
My bad. Corrected.